acvm-repo/acvm/tests/solver.rs

```diff
@@ -796,3 +797,61 @@ fn binary_operations() {
 
 }
 
+fn and_op() -> BlackBoxFuncCall<FieldElement> {
+    BlackBoxFuncCall::AND {
+        lhs: FunctionInput::witness(Witness(1), FieldElement::max_num_bits()),
+        rhs: FunctionInput::witness(Witness(2), FieldElement::max_num_bits()),
+        output: Witness(3),
+    }
+}
+
+fn xor_op() -> BlackBoxFuncCall<FieldElement> {
+    BlackBoxFuncCall::XOR {
+        lhs: FunctionInput::witness(Witness(1), FieldElement::max_num_bits()),
+        rhs: FunctionInput::witness(Witness(2), FieldElement::max_num_bits()),
+        output: Witness(3),
+    }
+}
+
+fn prop_assert_eq(op: BlackBoxFuncCall<FieldElement>, x: u128, y: u128) -> TestCaseResult {
+    let assertion: Result<_, TestCaseError> = prop_assert_eq!(solve_blackbox_func_call(op.clone(), x, y), solve_blackbox_func_call(op, y, x));
+    assertion?;
+    Ok(())
+}
+
+
+proptest! {
+
+    #[test]
+    fn and_commutative(x in any::<u128>(), y in any::<u128>()) {
```

Comment:
```suggestion
    fn and_commutative(x: u128, y:u128) {
```

If you're doing `any::<T>()` you can just use `: T` instead.